### PR TITLE
Integrating endpoints for User Profiles

### PR DESCRIPTION
**Pull request by VictorAJ01**

### CODE REVIEW

**Review by sammy-code98 — CHANGES_REQUESTED** [review]

File: components/Dashboard/candidate/profile/index.tsx [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
-
+import { useSession } from "next-auth/react";
+import { getCandidateProfile } from "@/api-endpoints/user-profile/user-profile.api";
+import { useQuery } from "@tanstack/react-query";
 
 const Candidate = (): JSX.Element => {
+    const { data: session } = useSession();
+    const jwtToken = session?.user?.accessToken as string;
+
+    const { data } = useQuery(["recruiterProfile"], async () => {
```
[the remark this answers is not in the document]
Comment (VictorAJ01) [reply components/Dashboard/candidate/profile/index.tsx]: Done

File: components/Dashboard/candidate/profile/index.tsx [hunk 2]
```diff
@@ -25,15 +34,15 @@ const Candidate = (): JSX.Element => {
                                 </span>
                             </div>
                             <div className="flex flex-col gap-2">
-                                <h3 className="text-lg font-bold">{userProfile.fullName}</h3>
+                                <h3 className="text-lg font-bold">{data?.first_name} {data?.last_name}</h3>
                                 <div className="flex items-center space-x-2">
                                     <BriefCase />
-                                    <p className="text-sm font-normal">{userProfile.job.title}</p>
+                                    <p className="text-sm font-normal">{data?.role ?? "Backend Engineer"}</p>
```
[the remark this answers is not in the document]
Comment (VictorAJ01) [reply components/Dashboard/candidate/profile/index.tsx]: Done

**Review by sammy-code98 — APPROVED** [review]

File: components/atoms/ToggleButton/index.tsx [hunk 3]
```diff
@@ -19,6 +21,7 @@ function ToggleButton({
           name="toggle"
           onChange={(e): React.ChangeEvent<HTMLInputElement> => onChange(e)}
           id="toggle"
+          defaultChecked={isChecked}
           className={`${styles.toggleCheckbox} border-2 border-gray-300 absolute block w-6 h-6 rounded-full bg-white appearance-none cursor-pointer`}
```
Comment (chimdie) [inline components/atoms/ToggleButton/index.tsx:25]:
```suggestion
          className={`${styles.toggleCheckbox} border-2 border-gray-300 absolute block w-6 h-6 rounded-full bg-white appearance-none cursor-pointer aspect-square`}
```

File: components/atoms/ToggleButton/index.tsx [hunk 4]
```diff
@@ -19,6 +21,7 @@ function ToggleButton({
           name="toggle"
           onChange={(e): React.ChangeEvent<HTMLInputElement> => onChange(e)}
```
Comment (chimdie) [inline components/atoms/ToggleButton/index.tsx:22]: can you create this function outside the template

File: components/Dashboard/recruiter/profile/index.tsx [hunk 5]
```diff
@@ -18,7 +28,7 @@ const Recruiter = (): JSX.Element => {
                         <div className="flex items-center space-x-4 relative">
                             <div className="relative w-20 h-20">
                                 <Image
-                                    src={`${pic9.src}`}
+                                    src={`${data?.profile_image ?? pic9.src}`}
                                     fill
                                     alt=""
                                     className="w-14 h-14 rounded-full"
```
Comment (chimdie) [inline components/Dashboard/recruiter/profile/index.tsx:34]:
```suggestion
                                    className="w-14 h-14 rounded-full aspect-square"
```

File: package-lock.json
Comment (chimdie) [inline package-lock.json:1]: can you delete this file, use yarn to reinstall all dependencies

**Review by chimdie — CHANGES_REQUESTED** [review]

File: components/atoms/ToggleButton/index.tsx (hunk 3 above)
Comment (VictorAJ01) [reply components/atoms/ToggleButton/index.tsx:25]: Done

File: components/Dashboard/recruiter/profile/index.tsx (hunk 5 above)
Comment (VictorAJ01) [reply components/Dashboard/recruiter/profile/index.tsx:34]: Done

File: package-lock.json
Comment (VictorAJ01) [reply package-lock.json:1]: Okay

Comment (VictorAJ01) [reply package-lock.json:1]: Done

**Review by chimdie — APPROVED** [review]
@sammy-code98 LGTM

**Comment by chimdie** [comment]
@VictorAJ01 Please fix the conflicts

**Review by chimdie — APPROVED** [review]
@copy-ninja1 @sammy-code98   Ready to be merged

**Pull request merged**